File: programmers/코딩테스트 입문 with Java/암호해독.java
package 코딩테스트입문;

/**
 * https://school.programmers.co.kr/learn/courses/30/lessons/120892
 */
public class 암호해독 {
    public static void main(String[] args) {
        System.out.println(solution("dfjardstddetckdaccccdegk",4));
        System.out.println(solution("pfqallllabwaoclk",2));
    }
    public static String solution(String cipher, int code) {
        String answer = "";
        for(int i=code-1; i<cipher.length();i+=code){
            answer+=cipher.charAt(i);
        }
        return answer;
    }
}

File: programmers/코딩테스트 입문 with Python/암호 해독.py
def solution(cipher, code):
    answer = ''
    # for c in range(code, len(cipher)+1, code):
    #     answer += cipher[c - 1]
    answer = cipher[code-1::code]
    return answer


print(solution("dfjardstddetckdaccccdegk", 4))
print(solution("pfqallllabwaoclk", 2))
